File: Ejercicio 3/institucion.py
class Persona:
    def __init__(self, nombre, identificacion, edad, correo):
        self.nombre = nombre
        self.identificacion = identificacion
        self.edad = edad
        self.correo = correo

    def mostrar_info(self):
        print("Nombre:", self.nombre)
        print("ID:", self.identificacion)
        print("Edad:", self.edad)
        print("Correo:", self.correo)

    def datos_extra(self):
        pass


class Estudiante(Persona):
    def __init__(self, nombre, identificacion, edad, correo, programa, semestre, codigo):
        super().__init__(nombre, identificacion, edad, correo)
        self.programa = programa
        self.semestre = semestre
        self.codigo = codigo

    def datos_extra(self):
        print("Programa:", self.programa)
        print("Semestre:", self.semestre)
        print("Código:", self.codigo)


class Docente(Persona):
    def __init__(self, nombre, identificacion, edad, correo, profesion, salario, materia):
        super().__init__(nombre, identificacion, edad, correo)
        self.profesion = profesion
        self.salario = salario
        self.materia = materia

    def datos_extra(self):
        print("Profesión:", self.profesion)
        print("Salario:", self.salario)
        print("Materia:", self.materia)


class Administrativo(Persona):
    def __init__(self, nombre, identificacion, edad, correo, cargo, departamento, salario):
        super().__init__(nombre, identificacion, edad, correo)
        self.cargo = cargo
        self.departamento = departamento
        self.salario = salario

    def datos_extra(self):
        print("Cargo:", self.cargo)
        print("Departamento:", self.departamento)
        print("Salario:", self.salario)



estudiante = Estudiante(
    "Ana Torres", "1001", 20, "ana@gmail.com",
    "Ingeniería de Sistemas", 4, "EST123"
)

docente = Docente(
    "Carlos Ruiz", "2002", 45, "carlos@gmail.com",
    "Ingeniero de sistemas", 3500000, "Programación"
)

admin = Administrativo(
    "Laura Gómez", "3003", 38, "laura@gmail.com",
    "Coordinadora", "Admisiones", 2800000
)

personas = [estudiante, docente, admin]

for p in personas:
    print("Tipo:", type(p).__name__)
    p.mostrar_info()
    p.datos_extra()
    print("-------------------")

File: Ejercicio 3/institucion.java
class Persona {
    String nombre;
    String identificacion;
    int edad;
    String correo;

    public Persona(String nombre, String identificacion, int edad, String correo) {
        this.nombre = nombre;
        this.identificacion = identificacion;
        this.edad = edad;
        this.correo = correo;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre);
        System.out.println("ID: " + identificacion);
        System.out.println("Edad: " + edad);
        System.out.println("Correo: " + correo);
    }

    public void datosExtra() {}
}


class Estudiante extends Persona {
    String programa;
    int semestre;
    String codigo;

    public Estudiante(String nombre, String identificacion, int edad, String correo,
                      String programa, int semestre, String codigo) {
        super(nombre, identificacion, edad, correo);
        this.programa = programa;
        this.semestre = semestre;
        this.codigo = codigo;
    }

    public void datosExtra() {
        System.out.println("Programa: " + programa);
        System.out.println("Semestre: " + semestre);
        System.out.println("Código: " + codigo);
    }
}


class Docente extends Persona {
    String profesion;
    double salario;
    String materia;

    public Docente(String nombre, String identificacion, int edad, String correo,
                   String profesion, double salario, String materia) {
        super(nombre, identificacion, edad, correo);
        this.profesion = profesion;
        this.salario = salario;
        this.materia = materia;
    }

    public void datosExtra() {
        System.out.println("Profesión: " + profesion);
        System.out.println("Salario: " + salario);
        System.out.println("Materia: " + materia);
    }
}


class Administrativo extends Persona {
    String cargo;
    String departamento;
    double salario;

    public Administrativo(String nombre, String identificacion, int edad, String correo,
                          String cargo, String departamento, double salario) {
        super(nombre, identificacion, edad, correo);
        this.cargo = cargo;
        this.departamento = departamento;
        this.salario = salario;
    }

    public void datosExtra() {
        System.out.println("Cargo: " + cargo);
        System.out.println("Departamento: " + departamento);
        System.out.println("Salario: " + salario);
    }
}


public class Main {
    public static void main(String[] args) {

        Persona estudiante = new Estudiante(
                "Ana Torres", "1001", 20, "ana@gmail.com",
                "Ingeniería de Sistemas", 4, "EST123"
        );

        Persona docente = new Docente(
                "Carlos Ruiz", "2002", 45, "carlos@gmail.com",
                "Ingeniero de sistemas", 3500000, "Programación"
        );

        Persona admin = new Administrativo(
                "Laura Gómez", "3003", 38, "laura@gmail.com",
                "Coordinadora", "Admisiones", 2800000
        );

        Persona[] personas = {estudiante, docente, admin};

        for (Persona p : personas) {
            System.out.println("Tipo: " + p.getClass().getSimpleName());
            p.mostrarInfo();
            p.datosExtra();
            System.out.println("-------------------");
        }
    }
}
